pkg/config/tenancy/tenancy.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2022 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tenancy
+
+// TenancyConfig holds the settings for multi-tenant Jaeger
+type TenancyConfig struct {
+	Enabled bool
+	Header  string
+	Valid   Guard
```

Comment:
this looks weird in the usage: `config.Valid.Valid()`. How about:
```
type TenancyConfig struct {
    ...
    guard Guard
}

func (c *TenancyConfig) Valid(v string) bool {
    return c.guard.Valid()
}
```

especially since the current Valid variable is force-assigned in the constructor & cannot be user-supplied.